api/routes/reviews.js

```diff
@@ -0,0 +1,86 @@
+const express = require('express');
+const router = express.Router();
+const Review = require('../models/review');
+const { hashId } = require('../utils/helper');
+
+// Getting all reviews
+router.get('/', async (req, res) => {
+  try {
+    const reviews = await Review.find();
+    res.json(reviews);
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// create a review
+router.post('/', async (req, res) => {
+  const review = new Review({
+    id: hashId(req.body.title + req.body.author + req.body.desc + req.body.created),
+    title: req.body.title,
+    rating: req.body.rating,
+    desc: req.body.desc,
+    lastEdited: (new Date()).now(),
+    created: (new Date()).now(), 
+    author: req.body.author,
+  });
+
+  try {
+    const newReview = await review.save();
+    res.status(201).json(newReview);
+  } catch (err) {
+    res.status(400).json({ message: err.message });
+  }
+});
+
+// Getting one review by id
+router.get('/:id', getReview, (req, res) => {
+  res.json(res.review);
+});
+
+// update review
+router.patch('/:id', getReview, async (req, res) => {
+  if (req.body.title != null) {
+    res.review.title = req.body.title;
+  }
+
+  if (req.body.rating != null) {
+    res.review.rating = req.body.rating;
+  }
+
+  if (req.body.desc != null) {
+    res.review.desc = req.body.desc;
+  }
+
+  if (req.body.lastEdited != null) {
```

Comment:
Remove this check for the body. We don't need them to provide a value to change `lastEdited`